cumulus/pallets/aura-ext/src/consensus_hook.rs

```diff
@@ -110,7 +121,7 @@ impl<
 	/// is more recent than the included block itself.
 	pub fn can_build_upon(included_hash: T::Hash, new_slot: Slot) -> bool {
 		let velocity = V.max(1);
-		let (last_slot, authored_so_far) = match pallet::SlotInfo::<T>::get() {
+		let (last_slot, authored_so_far) = match pallet::RelaySlotInfo::<T>::get() {
```

Comment:
I think can this be affected by https://github.com/paritytech/polkadot-sdk/issues/64, which leads to reading corrupted data here.

if block N updates the runtime, the code is overwritten immediately. The migration is run at the beginning of block N+1. Therefore, when we'll call `can_build_upon` at the end of block N, we'll use the new code with the old state

Reply:
It will not read corrupted state, as `RelaySlotInfo` doesn't exist yet.